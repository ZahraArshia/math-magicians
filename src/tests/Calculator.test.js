import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';

it('Calculator renders correctly', () => {
  const tree = render(<Calculator />);
  expect(tree).toMatchSnapshot();
});

it('Calculator user interaction', () => {
  render(
    <BrowserRouter>
      <Calculator />
    </BrowserRouter>,
  );
  const Element = screen.getByText('Lets do some math!');
  expect(Element).toBeInTheDocument();
});
