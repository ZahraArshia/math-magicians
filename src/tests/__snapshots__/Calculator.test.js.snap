// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="layout"
      >
        <h2>
          Lets do some math!
        </h2>
        <div
          class="skeleton"
        >
          <div
            class="screen"
          >
            0
          </div>
          <div
            class="keyboard"
          >
            <button
              class="gray"
              type="button"
            >
              AC
            </button>
            <button
              class="gray"
              type="button"
            >
              +/-
            </button>
            <button
              class="gray"
              type="button"
            >
              %
            </button>
            <button
              class="orange"
              type="button"
            >
              รท
            </button>
            <button
              class="gray"
              type="button"
            >
              7
            </button>
            <button
              class="gray"
              type="button"
            >
              8
            </button>
            <button
              class="gray"
              type="button"
            >
              9
            </button>
            <button
              class="orange"
              type="button"
            >
              x
            </button>
            <button
              class="gray"
              type="button"
            >
              4
            </button>
            <button
              class="gray"
              type="button"
            >
              5
            </button>
            <button
              class="gray"
              type="button"
            >
              6
            </button>
            <button
              class="orange"
              type="button"
            >
              -
            </button>
            <button
              class="gray"
              type="button"
            >
              1
            </button>
            <button
              class="gray"
              type="button"
            >
              2
            </button>
            <button
              class="gray"
              type="button"
            >
              3
            </button>
            <button
              class="orange"
              type="button"
            >
              +
            </button>
            <button
              class="gray"
              id="zero"
              type="button"
            >
              0
            </button>
            <button
              class="gray"
              type="button"
            >
              .
            </button>
            <button
              class="orange"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="layout"
    >
      <h2>
        Lets do some math!
      </h2>
      <div
        class="skeleton"
      >
        <div
          class="screen"
        >
          0
        </div>
        <div
          class="keyboard"
        >
          <button
            class="gray"
            type="button"
          >
            AC
          </button>
          <button
            class="gray"
            type="button"
          >
            +/-
          </button>
          <button
            class="gray"
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
            รท
          </button>
          <button
            class="gray"
            type="button"
          >
            7
          </button>
          <button
            class="gray"
            type="button"
          >
            8
          </button>
          <button
            class="gray"
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
          <button
            class="gray"
            type="button"
          >
            4
          </button>
          <button
            class="gray"
            type="button"
          >
            5
          </button>
          <button
            class="gray"
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
          <button
            class="gray"
            type="button"
          >
            1
          </button>
          <button
            class="gray"
            type="button"
          >
            2
          </button>
          <button
            class="gray"
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
          <button
            class="gray"
            id="zero"
            type="button"
          >
            0
          </button>
          <button
            class="gray"
            type="button"
          >
            .
          </button>
          <button
            class="orange"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
